device/pcie/pci_device.cpp

```diff
@@ -23,6 +23,38 @@
 #include "common/assert.hpp"
 #include "common/logger.hpp"
 
+static PciDeviceInfo read_device_info(int fd)
+{
+    tenstorrent_get_device_info info{};
+    info.in.output_size_bytes = sizeof(info.out);
+
+    if (ioctl(fd, TENSTORRENT_IOCTL_GET_DEVICE_INFO, &info) < 0) {
+        TT_THROW("TENSTORRENT_IOCTL_GET_DEVICE_INFO failed");
+    }
+
+    uint16_t bus = info.out.bus_dev_fn >> 8;
+    uint16_t dev = (info.out.bus_dev_fn >> 3) & 0x1F;
+    uint16_t fn = info.out.bus_dev_fn & 0x07;
+
+    return PciDeviceInfo{info.out.vendor_id, info.out.device_id, info.out.pci_domain, bus, dev, fn};
+}
+
+static int determine_numa_node(int fd)
+{
+    const auto device_info = read_device_info(fd);
```

Comment:
You will already have this in this->info ?

Reply:
I think I will sort this out in #147